From 6dc327ae77fb02af2b23584776c843753947a911 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Wed, 1 Apr 2020 18:28:54 +0200
Subject: [PATCH 1/1] queueing: skb_reset_redirect change has been backported
 to 5.[45]

This is a follow up to 2d4fa2a6e7903ec3340f1b075456cbd84ba6a744.

Upstream commit 2c64605b590edadb3fb46d1ec6badb49e940b479 has been backported
to 5.4.29 and 5.5.14.

Signed-off-by: Christian Hesse <mail@eworm.de>
---
 src/compat/compat.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/compat/compat.h b/src/compat/compat.h
index fe2d07e..7f16332 100644
--- a/src/compat/compat.h
+++ b/src/compat/compat.h
@@ -1024,7 +1024,8 @@ out:
 #define COMPAT_CANNOT_USE_MAX_MTU
 #endif
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 6, 0)
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 29) || \
+    (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0) && LINUX_VERSION_CODE < KERNEL_VERSION(5, 5, 14)))
 #include <linux/skbuff.h>
 static inline void skb_reset_redirect(struct sk_buff *skb)
 {
